### PR TITLE
Use Documentation Tags for Maven and Gradle in Getting Started

### PR DESCRIPTION
**Pull request by harpreets789**
This PR consolidates the separate XML and Gradle code snippets into joint tabs within the getting-spring-security.adoc file
Issue gh-16228

### CODE REVIEW

File: docs/modules/ROOT/pages/getting-spring-security.adoc [hunk 1]
```diff
@@ -15,19 +15,22 @@ Typically, these are done to provide improved security to match modern security
 
 
 [[maven]]
```
Comment (jzheaux) [inline docs/modules/ROOT/pages/getting-spring-security.adoc:17]: Please add `[[getting-gradle]]` so that bookmarks to this anchor still work

File: docs/modules/ROOT/pages/getting-spring-security.adoc [hunk 2]
```diff
@@ -15,19 +15,22 @@ Typically, these are done to provide improved security to match modern security
...
-== Usage with Maven
+== Usage with Maven and Gradle
```
Comment (jzheaux) [inline docs/modules/ROOT/pages/getting-spring-security.adoc:18]: I think `Usage` is just fine at this point.

File: docs/modules/ROOT/pages/getting-spring-security.adoc [hunk 3]
```diff
@@ -15,19 +15,22 @@ Typically, these are done to provide improved security to match modern security
...
 
-As most open source projects, Spring Security deploys its dependencies as Maven artifacts.
-The topics in this section describe how to consume Spring Security when using Maven.
+As most open source projects, Spring Security deploys its dependencies as Maven artifacts, which makes them compatible with both Maven and Gradle. The following sections demonstrate how to integrate Spring Security with these build tools, with examples for Spring Boot and standalone usage.
 
 [[getting-maven-boot]]
-=== Spring Boot with Maven
+[[getting-gradle-boot]]
+=== Spring Boot with Maven and Gradle
```
Comment (jzheaux) [inline docs/modules/ROOT/pages/getting-spring-security.adoc:24]: Since you've already stated that you'll be talking about both, it's okay to simply say `Spring Boot`.

File: docs/modules/ROOT/pages/getting-spring-security.adoc [hunk 4]
```diff
@@ -40,9 +43,25 @@ Alternatively, you can manually add the starter, as the following example shows:
...
+----
+dependencies {
+	implementation "org.springframework.boot:spring-boot-starter-security"
+}
+----
+======
+
 Since Spring Boot provides a Maven BOM to manage dependency versions, you do not need to specify a version.
-If you wish to override the Spring Security version, you can do so by providing a Maven property:
+If you wish to override the Spring Security version, you can do so by providing a Maven property or declaring it as a Gradle property, as shown below:
```
Comment (jzheaux) [inline docs/modules/ROOT/pages/getting-spring-security.adoc:59]: Let's see if we can keep it still more generic to simplify the wording. Maybe:

```
If you wish to override the Spring Security version, you can do so with a build property as shown below:
```

File: docs/modules/ROOT/pages/getting-spring-security.adoc [hunk 5]
```diff
@@ -52,10 +71,24 @@ If you wish to override the Spring Security version, you can do so by providing
...
+[subs="verbatim,attributes"]
+----
+ext['spring-security.version']='{spring-security-version}'
+----
+======
+
 Since Spring Security makes breaking changes only in major releases, you can safely use a newer version of Spring Security with Spring Boot.
 However, at times, you may need to update the version of Spring Framework as well.
-You can do so by adding a Maven property:
+You can do so by adding a Maven or Gradle property:
```
Comment (jzheaux) [inline docs/modules/ROOT/pages/getting-spring-security.adoc:86]: Let's consider the same simplification here:

```
You can do so by adding a build property like so:
```

File: docs/modules/ROOT/pages/getting-spring-security.adoc [hunk 6]
```diff
@@ -65,13 +98,27 @@ You can do so by adding a Maven property:
...
+======
+
 If you use additional features (such as LDAP, OAuth 2, and others), you need to also include the appropriate xref:modules.adoc#modules[Project Modules and Dependencies].
 
 [[getting-maven-no-boot]]
-=== Maven Without Spring Boot
+=== Standalone Usage (Without Spring Boot)
 
-When you use Spring Security without Spring Boot, the preferred way is to use Spring Security's BOM to ensure that a consistent version of Spring Security is used throughout the entire project. The following example shows how to do so:
+When you use Spring Security without Spring Boot, the preferred way is to use Spring Security's BOM to ensure that a consistent version of Spring Security is used throughout the entire project.  For Gradle, you can do so by using the https://github.com/spring-gradle-plugins/dependency-management-plugin[Dependency Management Plugin]:
```
Comment (jzheaux) [inline docs/modules/ROOT/pages/getting-spring-security.adoc:116]: I'd recommend leaving this link to the plugin until after the snippet and also inside a `TIP`:

```
[TIP]
Spring provides a `Dependency Management Plugin`(link) for Gradle
```

I'd recommend doing the same with the same link later on in the article.

File: docs/modules/ROOT/pages/getting-spring-security.adoc [hunk 7]
```diff
@@ -138,12 +243,24 @@ This approach uses Maven's "`bill of materials`" (BOM) concept and is only avail
 For additional details about how dependencies are resolved, see https://maven.apache.org/guides/introduction/introduction-to-dependency-mechanism.html[Maven's Introduction to the Dependency Mechanism documentation].
 ====
 
-[[maven-repositories]]
-=== Maven Repositories
-All GA releases (that is, versions ending in .RELEASE) are deployed to Maven Central, so you need not declare additional Maven repositories in your pom.
+[[maven-gradle-gearepositories]]
+=== Maven and Gradle Repositories
```
Comment (jzheaux) [inline docs/modules/ROOT/pages/getting-spring-security.adoc:247]: I think this should say as "Maven Repositories" since that's what they are in either case.

File: docs/modules/ROOT/pages/getting-spring-security.adoc [hunk 8]
```diff
@@ -138,12 +243,24 @@ This approach uses Maven's "`bill of materials`" (BOM) concept and is only avail
...
+All GA releases (that is, versions ending in .RELEASE) are deployed to Maven Central, so you need not declare additional Maven repositories in your `pom.xml` or, for Gradle, using the `mavenCentral()` repository is sufficient for GA releases.
```
Comment (jzheaux) [inline docs/modules/ROOT/pages/getting-spring-security.adoc:248]: Please place the "For Gradle" in a separate sentence and separate paragraph. This makes the sentence shorter and continue to be about one thing.

**Comment by harpreets789** [comment]
Thank you for your thorough review and valuable suggestions. I appreciate the directions you've provided and will make the necessary changes accordingly.

**Comment by harpreets789** [comment]
> @jzheaux  I have addressed the PR review feedback and made the necessary updates to the "getting-spring-security" documentation. Please let me know if there are any additional changes required!

**Review by jzheaux — APPROVED** [review]

**Comment by jzheaux** [comment]
Thanks, @harpreets789! This is now merged into `main`.
